examples/platform/openiotsdk/app/openiotsdk_platform.cpp

```diff
@@ -188,6 +189,13 @@ int openiotsdk_platform_init(void)
         return EXIT_FAILURE;
     }
 
+    ret = psa_crypto_init();
```

Comment:
I thought about having a define, but it works both with and without psa being available but we can certainly try and guard it anyway.

Reply:
Yeah, it works because we set TF-M support by default to each example. But let's guard it in case that ever changes.

Reply:
I would suggest
#if defined(MBEDTLS_PSA_CRYPTO_C)
as this is where similar code is guarded elsewhere and that code is part of the mbedtls PSA still.

Reply:
Your suggestion looks good to me.

Reply:
I'm not sure whether the dependency on mbedtls configuration is correct at this point. I think about something like that: 
```suggestion
#if (CONFIG_CHIP_CRYPTO == "psa")
    ret = psa_crypto_init();
#endif
```
Then we need to pass the `CONFIG_CHIP_CRYPTO' flag to `openiotsdk-app` target. The same as we did with TF-M support: https://github.com/ARM-software/connectedhomeip/blob/89039726d54eab72526aa66868779fa29d465351/config/openiotsdk/CMakeLists.txt#L288

Reply:
I'm not sure I agree at this point. The current implementation of PSA relies on some underlying calls to mbedtls still. Those calls themselves are guarded by MBEDTLS_PSA_CRYPTO_C so that make far more sense to me. If in the future the PSA implementation moves away from its reliance on mbedtls then I think your suggestion then makes more sense. The logic goes like this, we could have CONFIG_CHIP_CRYPTO set to psa but MBEDTLS_PSA_CRYPTO_C set False and it wouldn't work....

Reply:
Sure, but let's look at the other side. If we have `MBEDTLS_PSA_CRYPTO_C` enabled in our mbedtls does it mean that we want to use PSA crypto in the application? If we build an application with mbedtls crypto we still initialize the psa_crypto because the mbedtls_config stays the same.

Reply:
Maybe we should have both, nested?
#if (CONFIG_CHIP_CRYPTO == "psa")
#if defined(MBEDTLS_PSA_CRYPTO_C)

??

Reply:
I think `#if (CONFIG_CHIP_CRYPTO == "psa")` should be enough.

Reply:
LGTM